modules/living_spaces_group/living_spaces_group.module

```diff
@@ -1239,12 +1239,18 @@ function living_spaces_group_group_update(GroupInterface $group) {
  */
 function living_spaces_group_query_living_spaces_menus_user_spaces_alter(Drupal\Core\Database\Query\AlterableInterface $query) {
   $query->leftJoin('user__joined_spaces', 'user__joined_spaces', 'user__joined_spaces.joined_spaces_target_id = groups_field_data.id');
-  $query->condition('user__joined_spaces.entity_id', \Drupal::currentUser()->id(), '=');
+
+  $or_condition = $query->orConditionGroup()
+    ->condition('user__joined_spaces.entity_id', \Drupal::currentUser()->id(), '=')
+    ->condition('groups_field_data.living_space_privacy', 'public', '=');
+
+  $query->condition($or_condition);
 
   $query->leftJoin('user__field_favorite_spaces', 'user__field_favorite_spaces', 'user__field_favorite_spaces.field_favorite_spaces_target_id = groups_field_data.id AND user__field_favorite_spaces.entity_id = user__joined_spaces.entity_id');
   $query->fields('user__field_favorite_spaces', ['delta', 'entity_id']);
   $query->addExpression('IFNULL(user__field_favorite_spaces.delta, 1000)', 'user__field_favorite_spaces_delta');
+  $query->addExpression('IFNULL(user__joined_spaces.entity_id, 1000)', 'user__joined_spaces_entity_id');
 
   $order_by = &$query->getOrderBy();
-  $order_by = array_merge(['user__field_favorite_spaces_delta' => 'ASC'], $order_by);
+  $order_by = array_merge(['user__field_favorite_spaces_delta' => 'ASC', 'user__joined_spaces_entity_id' => 'DESC'], $order_by);
```

Comment:
This will cause an issue and I have no idea why it have been added.
It will list all spaces user ARE NOT member of first, which is exactly opposite of what it should be doing (i.e. only displaying space where user is member)

Reply:
Done!